### PR TITLE
BUG: ImageSpatialObject's BoundingBox broken if not axis aligned

### PR DESCRIPTION
**Pull request by HastingsGreer**

### CODE REVIEW

File: Modules/Core/SpatialObjects/include/itkImageMaskSpatialObject.hxx [hunk 1]
```diff
@@ -51,25 +50,24 @@ ImageMaskSpatialObject< TDimension >
...
-          NumericTraits<PixelType>::ZeroValue() ) );
-      if( insideMask )
-        {
-        return true;
-        }
+      bool inside_image = m_Image->TransformPhysicalPointToIndex( p,
+      index );
+      if inside_image {
+          using InterpolatorOutputType = typename InterpolatorType::OutputType;
+          const bool insideMask = (
```
Comment (floryst) [inline Modules/Core/SpatialObjects/include/itkImageMaskSpatialObject.hxx]: quoi?

Comment (aylward) [inline Modules/Core/SpatialObjects/include/itkImageMaskSpatialObject.hxx]: const?

File: Modules/Core/SpatialObjects/include/itkImageSpatialObject.hxx [hunk 2]
```diff
@@ -191,6 +194,13 @@ ImageSpatialObject< TDimension,  PixelType >
   return true;
 }
 
+template< unsigned int TDimension, typename PixelType>
+void
+ImageSpatialObject< TDimension, PixelType >
+::Update()
+{
+    Superclass::Update();
+}
```
Comment (aylward) [inline Modules/Core/SpatialObjects/include/itkImageSpatialObject.hxx]: This member function can be removed.

**Comment by floryst** [comment]
LGTM

**Pull request merged**